src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthRpcModuleTests.EthCall.cs

```diff
@@ -192,24 +212,106 @@ public async Task Eth_call_without_gas_pricing_after_1559_legacy()
-                .Op(Instruction.BASEFEE)
-                .PushData(0)
-                .Op(Instruction.SSTORE)
-                .Done;
+        }
+        
+        /// <summary>
+        /// 5. After 1559: In geth if you do specify a gas price, that is interpreted as both the max and priority fee
+        /// and your account balance is checked against them + the current base fee.
+        /// </summary>
+        /// TODO maybe needs to be changed to ``geth``
+        [Test]
+        public async Task Eth_call_with_gas_pricing_after_1559_new_type_of_transaction()
+        {
+            using Context ctx = await Context.CreateWithLondonEnabled();
+            TransactionForRpc transaction = ctx._test.JsonSerializer.Deserialize<TransactionForRpc>(
+                "{\"jsonrpc\":\"2.0\",\"error\":{\"code\":-32015,\"message\":\"VM execution error.\",\"data\":\"insufficient sender balance\"},\"id\":67}");
+            string serialized = ctx._test.TestEthRpc("eth_call", ctx._test.JsonSerializer.Serialize(transaction));
+            Assert.AreEqual("{\"jsonrpc\":\"2.0\",\"result\":\"0x\",\"id\":67}", serialized);
+        }
+        
+        [Test]
+        public async Task Eth_call_with_gas_limit_after_1559_new_type_of_transaction()
+        {
+            using Context ctx = await Context.CreateWithLondonEnabled();
+            TransactionForRpc transaction = ctx._test.JsonSerializer.Deserialize<TransactionForRpc>(
+                "{\"from\": \"0x32e4e4c7c5d1cea5db5f9202a9e4d99e56c91a24\", \"to\": \"0x32e4e4c7c5d1cea5db5f9202a9e4d99e56c91a24\", \"gas\": \"0x1000\"}");
+            string serialized = ctx._test.TestEthRpc("eth_call", ctx._test.JsonSerializer.Serialize(transaction));
+            Assert.AreEqual("{\"jsonrpc\":\"2.0\",\"error\":{\"code\":-32015,\"message\":\"VM execution error.\",\"data\":\"gas limit below intrinsic gas\"},\"id\":67}", serialized);
+        }
+        
+        /// <summary>
+        /// 6. After 1559: If you specify a gas price of type 1559 and transaction type 0x2,
+        /// the transaction will be of type 1559 and your account balance is checked against them + the current basefee.
+        /// (maxFeePerGas)
+        /// </summary>
+        [Test]
+        public async Task Eth_call_with_max_fee_after_1559_new_type_of_transaction()
+        {
+            using Context ctx = await Context.CreateWithLondonEnabled(); 
+            TransactionForRpc transaction = ctx._test.JsonSerializer.Deserialize<TransactionForRpc>(
+                "{\"from\": \"0x32e4e4c7c5d1cea5db5f9202a9e4d99e56c91a24\", \"to\": \"0x32e4e4c7c5d1cea5db5f9202a9e4d99e56c91a24\", \"maxFeePerGas\": \"0x2DA2830C\", \"value\": \"0x2DA2830C\"}");
+            string serialized = ctx._test.TestEthRpc("eth_call", ctx._test.JsonSerializer.Serialize(transaction));
+            Assert.AreEqual("{\"jsonrpc\":\"2.0\",\"error\":{\"code\":-32015,\"message\":\"VM execution error.\",\"data\":\"insufficient sender balance\"},\"id\":67}", serialized);
```

Comment:
Could we return errors like in geth?
- max fee per gas less than block base fee
- max priority fee per gas higher than max fee per gas